components/server/ee/src/gitea/gitea-app-support.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+ * Licensed under the Gitpod Enterprise Source Code License,
+ * See License.enterprise.txt in the project root folder.
+ */
+
+import { AuthProviderInfo, ProviderRepository, User } from "@gitpod/gitpod-protocol";
+import { inject, injectable } from "inversify";
+import { TokenProvider } from "../../../src/user/token-provider";
+import { UserDB } from "@gitpod/gitpod-db/lib";
+import { Gitea } from "../../../src/gitea/api";
+
+@injectable()
+export class GiteaAppSupport {
+
+    @inject(UserDB) protected readonly userDB: UserDB;
+    @inject(TokenProvider) protected readonly tokenProvider: TokenProvider;
+
+    async getProviderRepositoriesForUser(params: { user: User, provider: AuthProviderInfo }): Promise<ProviderRepository[]> {
+        const token = await this.tokenProvider.getTokenForHost(params.user, params.provider.host);
+        const oauthToken = token.value;
+        const api = Gitea.create(`https://${params.provider.host}`, oauthToken);
```

Comment:
Two tiny nitpicks, 1. It is possible for gitea to exist on a subpath (eg. https://example.com/gitea/ etc..), and 2. some users may wish not to use https (not a great idea, but one that people make for a variety of reasons).

You may wish instead of host, to use something like `baseURL` where users can fill the url scheme and full url (incl. a sub-path if they so require).

Reply:
Good points. I will have a look and do some tests later. Currently my main problem is still getting a Gitpod instance with those changes up and running. 🙈

Reply:
Is the issue a matter of compute resources available to you, or being able to apply your changes to a running deploy? If the former, the Gitea project likely can assist in providing compute resources, if the later I'm sure if you post the issue you are running into then someone can help.

Reply:
I currently using a cluster on vultr with with some credit I had lying around, so should be fine for the beginning, but I will come back to your offer if needed. Main problem is building images of the needed components and deploying those atm. It requires some manual intervention which loads to a pretty slow dev cycle as I have to rebuild the server, deploy it and look for changes ... The gitpod team seems to use telepresence for  that by am not quite sure how to integrate that with my own cluster.